From a869776027af711358c268d7cd4032528f310321 Mon Sep 17 00:00:00 2001
From: Andrey Grodzovsky <andrey.grodzovsky@amd.com>
Date: Mon, 24 Sep 2018 14:10:22 +0200
Subject: [PATCH 5448/5725] drm/amdgpu: Deactivate SW interrupt fallback in
 amdgpu_fence_process v2

Deactivate SW interrupt fallback when all emited fences are completed.
Also switch interrupt SW fallback message from INFO to WARN.

v2: shorten the warnign message a bit and only re-activate the timer during
processing if it was already activated before.

Signed-off-by: Andrey Grodzovsky <andrey.grodzovsky@amd.com>
Suggested-by: Christian Konig <Christian.Koenig@amd.com>
Signed-off-by: Raveendra Talabattula <raveendra.talabattula@amd.com>
---
 drivers/gpu/drm/amd/amdgpu/amdgpu_fence.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu_fence.c b/drivers/gpu/drm/amd/amdgpu/amdgpu_fence.c
index a9b4684..d169840 100644
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_fence.c
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_fence.c
@@ -274,7 +274,7 @@ static void amdgpu_fence_fallback(struct timer_list *t)
 	struct amdgpu_ring *ring = from_timer(ring, t,
 					      fence_drv.fallback_timer);
 
-	DRM_INFO("Fallback to SW interrupt on ring %s due to HW interrupt time out", ring->name);
+	DRM_WARN("Fence fallback timer expired on ring %s\n", ring->name);
 	amdgpu_fence_process(ring);
 }
 
-- 
2.7.4

